ts/navigation/MainNavigator.tsx

```diff
@@ -58,17 +65,27 @@ const navigation = TabNavigator(
   },
   {
     navigationOptions: ({ navigation: nav }) => ({
-      tabBarIcon: ({ focused }) => {
+      tabBarLabel: _ => {
+        const { routeName } = nav.state;
+        return ROUTE_LABEL[routeName as RoutesWithIcons];
```

Comment:
what's the type of `routeName`, why it needs a cast? is there a safeguard that avoids looking up non existent keys?

Reply:
`routeName`'s type is [`string`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/55933199f8734b7f79ab74330d48fa0aa612ba81/types/react-navigation/index.d.ts#L91), so the cast is needed, hope this addresses your question (will add this in a comment).

`RoutesWithIcons` (which I should probably rename) is defined as the union of the 5 routes that are provided to the tab navigator. Here, I'd like to have something like the following: 

```
const routes = {
 [ROUTES.ROUTE_A]: { screen: ... },
 [ROUTES.ROUTE_B]: { screen: ... },
 ...
};
type RoutesWithIcons = keyof typeof routes;
const navigation = TabNavigator(routes, ...);
```
But that does not behave as I would expect (it evaluates to `string`), nor does
```
type RoutesWithIcons = ROUTES.ROUTE_A | ROUTES.ROUTE_B  | ...
```
So my current definition for `RoutesWithIcons` is:
```
type RoutesWithIcons = "ROUTE_A" | "ROUTE_B" | ...;
```
Which I definitely do not like, and I should definitely figure out a better way to do that.

Anyway, this would just improve the definition of `RoutesWithIcons`, but the original problem would still exist. Should I add a extractor function (e.g. `getLabel(routeName: string): string`) with a fallback option?